pkg/controller/cassandradatacenter/statefulset.go

```diff
@@ -132,7 +132,7 @@ func newStatefulSetSpec(
 	podRackLabels := RackLabels(cdc, rack)
 	podLabels := PodTemplateSpecLabels(cdc)
 	statefulSetSpec := &v1beta2.StatefulSetSpec{
-		ServiceName: "cassandra", // TODO: correct service name? this service should already exist (apparently)
+		ServiceName: "cassandra-" + cdc.Name + "-nodes",
```

Comment:
without this, we could not even ping other pods from the pod so hostname resolution was not working. `ServiceName` field has to be same as the nodes service. This is a common mistake people do and they wonder why they cant ping stuff.